Exercise_4.py

```diff
@@ -8,16 +10,33 @@ def __init__(self, data):
           
 """ Inorder traversal of a binary tree"""
 def inorder(temp): 
-  
-    
-  
+    if temp:
+        inorder(temp.left)
+        print(temp.key, end=" ")
+        inorder(temp.right)
   
 """function to insert element in binary tree """
-def insert(temp,key): 
-  
-    
+def insert(temp,key):
+    if not temp:
+        root = newNode(key)
+        return root
+
+    ls = []
```

Comment:
Please use a queue or a deque instead, pop(0) causes shifting and compromises Time Complexity.

Reply:
Okay, thank you for the suggestions. I will make the required changes.